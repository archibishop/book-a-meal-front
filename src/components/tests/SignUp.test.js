import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import ConnectedSignUp, { SignUp } from '../auth/SignUp';

const mockStore = configureStore();
const initialState = {
    selectReducer: {
        selectedAvatar: 0,
    },
    login: (x) => {
    },
    user: [
        {
            message: 'HipHIP',
            error: null,
        }
    ],
};
const store = mockStore(initialState);

describe('<SignUp />', () => {
    describe('render()', () => {
        test('renders the component', () => {
            const wrapper = shallow(<ConnectedSignUp store={store} />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});

describe('<SignUp />', () => {
    describe('render()', () => {
        test('renders the component', () => {
            const wrapper = shallow(<SignUp />);
        });
    });
});


describe('<SignUp />', () => {
    describe('render()', () => {
        test('Test form submission', () => {
            const historyMock = jest.fn();
            const wrapper = shallow(<ConnectedSignUp store={store} history={historyMock}/>);

            expect(wrapper.find('form').length).toEqual(1)


            wrapper.find('form').simulate('click', {
                preventDefault: () => {
                },
                target: {
                    value: "caterer"
                }
            });

            wrapper.find('form').simulate('click', {
                preventDefault: () => {
                },
                target: {
                    value: "user"
                }
            });
        });
    });
});
