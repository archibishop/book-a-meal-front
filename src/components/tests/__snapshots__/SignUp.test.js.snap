// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> render() renders the component 1`] = `
<div>
  <Nav />
  <_class
    options={Object {}}
  />
  <form
    className="select-signup"
    onClick={[Function]}
  >
    <input
      name="gender"
      type="radio"
      value="user"
    />
     User
    <input
      name="gender"
      type="radio"
      value="caterer"
    />
     Caterer
  </form>
  <Connect(User)
    show={true}
  />
  <Connect(Caterer)
    show={true}
  />
</div>
`;
